apollo-federation/src/operation/mod.rs

```diff
@@ -3728,25 +3100,123 @@ impl Display for InlineFragment {
     }
 }
 
-impl Display for FragmentSpread {
-    fn fmt(&self, f: &mut Formatter<'_>) -> std::fmt::Result {
-        let data = self;
-        f.write_str("...")?;
-        f.write_str(&data.fragment_name)?;
-        data.directives.serialize().no_indent().fmt(f)
-    }
-}
-
 impl Display for OperationElement {
     fn fmt(&self, f: &mut Formatter<'_>) -> std::fmt::Result {
         match self {
             OperationElement::Field(field) => field.fmt(f),
             OperationElement::InlineFragment(inline_fragment) => inline_fragment.fmt(f),
-            OperationElement::FragmentSpread(fragment_spread) => fragment_spread.fmt(f),
         }
     }
 }
 
+/// Holds normalized selection sets of provided fragments.
+#[derive(Default)]
+pub(crate) struct FragmentSpreadCache {
+    fragment_selection_sets: Arc<HashMap<Name, SelectionSet>>,
```

Comment:
It is worth noting that we are dropping directives on the declaration of named fragments.
That wasn't used before. But, we may want to keep that comment somewhere.
```rust
        // PORT_NOTE: The JS codebase combined the fragment spread's directives with the fragment
        // definition's directives. This was invalid GraphQL as those directives may not be applicable
        // on different locations. While we now keep track of those references, they are currently ignored.
        pub(crate) fragment_directives: DirectiveList,
```